src/filters/filter_cue.cpp

```diff
@@ -301,11 +301,11 @@ namespace dmxfish::filters {
             float_channels.at(ind) = (end_value - start_value) * rel_time + start_value;
         } else if constexpr(std::is_same<T, dmxfish::dmx::pixel>::value)
         {
-            color_channels.at(ind) = dmxfish::dmx::pixel((end_value.hue - start_value.hue) * rel_time + start_value.hue,
-                                                         (end_value.saturation - start_value.saturation) * rel_time +
-                                                         start_value.saturation,
-                                                         (end_value.iluminance - start_value.iluminance) * rel_time +
-                                                         start_value.iluminance);
+            color_channels.at(ind) = dmxfish::dmx::pixel((uint16_t)((end_value.getRed() - start_value.getRed()) * rel_time + start_value.getRed()),
+                                                         (uint16_t)((end_value.getGreen() - start_value.getGreen()) * rel_time +
+                                                         start_value.getGreen()),
+                                                         (uint16_t)((end_value.getBlue() - start_value.getBlue()) * rel_time +
+                                                         start_value.getBlue()));
```

Comment:
is here the same problem?
color_channels is a vector. does the new object gets directly written in the vector?

Reply:
The vector contains that object. As the variables are part of it, no new object is created and the vector not modified.

Reply:
Not the object itself but its content. The vector contains the objects and the objects modify themself.